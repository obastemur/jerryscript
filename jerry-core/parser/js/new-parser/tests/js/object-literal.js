/* This test contains object literal parser tests. */

next_statement;

if ({},{/**/},{//
}) x++

next_statement;

var get = {member:"exp", "set" /**/ : //
 a + b, /**/ 'get' /**/ : /**/ c // comment
}

next_statement;

var set = {get:a?b:c, set : value, }

next_statement;

({ /*
*/  function : function () {}
 , get get() // comment
 {}, set set(arg) {},
 get/**/"get"/**/()
 {}, })

next_statement;

({/**/.3/**/:/**/0x09Af/**/,
  12e-1 : 14., 0XFaBe0189 : { 0e+3 : 0.077 + 9074 } })

next_statement;
