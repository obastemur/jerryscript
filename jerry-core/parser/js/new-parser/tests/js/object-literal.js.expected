
--- Script parsing start ---

  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT
  [  0] CBC_POP
  [  1] CBC_CREATE_OBJECT
  [  0] CBC_POP
  [  1] CBC_CREATE_OBJECT
  [  0] CBC_BRANCH_IF_FALSE_FORWARD
  [  1] CBC_POST_INCR_IDENT_PUSH_RESULT idx:1->ident(x)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT
  [  2] CBC_PUSH_LITERAL idx:4->string(exp)
  [  1] CBC_SET_PROPERTY idx:3->string(member)
  [  2] CBC_ADD_TWO_LITERALS idx:6->ident(a) idx:7->ident(b)
  [  1] CBC_SET_PROPERTY idx:5->string(set)
  [  2] CBC_PUSH_IDENT idx:9->ident(c)
  [  1] CBC_SET_PROPERTY idx:8->string(get)
  [  0] CBC_ASSIGN_IDENT idx:2->var_ident(get)
  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT
  [  2] CBC_PUSH_IDENT idx:6->ident(a)
  [  1] CBC_BRANCH_IF_FALSE_FORWARD
  [  2] CBC_PUSH_IDENT idx:7->ident(b)
  [  2] CBC_JUMP_FORWARD
  [  2] CBC_PUSH_IDENT idx:9->ident(c)
  [  1] CBC_SET_PROPERTY idx:8->string(get)
  [  2] CBC_PUSH_IDENT idx:11->ident(value)
  [  1] CBC_SET_PROPERTY idx:5->string(set)
  [  0] CBC_ASSIGN_IDENT idx:10->var_ident(set)
  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Literal encoding: small
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_UNDEFINED

Byte code size: 1 bytes

--- Function parsing end ---

  [  2] CBC_PUSH_LITERAL idx:13->function
  [  1] CBC_SET_PROPERTY idx:12->string(function)

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Literal encoding: small
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_UNDEFINED

Byte code size: 1 bytes

--- Function parsing end ---

  [  2] CBC_PUSH_LITERAL idx:14->function
  [  1] CBC_EXT_SET_GETTER idx:8->string(get)

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 1
  Literal encoding: small
  Argument range end: 1
  Register range end: 1
  Identifier range end: 1
  Const literal range end: 1
  Literal range end: 1

   0 : CBC_RETURN_WITH_UNDEFINED

Byte code size: 1 bytes

--- Function parsing end ---

  [  2] CBC_PUSH_LITERAL idx:15->function
  [  1] CBC_EXT_SET_SETTER idx:5->string(set)

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Literal encoding: small
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_UNDEFINED

Byte code size: 1 bytes

--- Function parsing end ---

  [  2] CBC_PUSH_LITERAL idx:16->function
  [  1] CBC_EXT_SET_GETTER idx:8->string(get)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT
  [  2] CBC_PUSH_LITERAL idx:18->number(0x09Af)
  [  1] CBC_SET_PROPERTY idx:17->number(.3)
  [  2] CBC_PUSH_LITERAL idx:20->number(14.)
  [  1] CBC_SET_PROPERTY idx:19->number(12e-1)
  [  2] CBC_CREATE_OBJECT
  [  3] CBC_ADD_TWO_LITERALS idx:23->number(0.077) idx:24->number(9074)
  [  2] CBC_SET_PROPERTY idx:22->number(0e+3)
  [  1] CBC_SET_PROPERTY idx:21->number(0XFaBe0189)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_IDENT idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK

Final byte code dump:

  Maximum stack depth: 3
  Literal encoding: small
  Argument range end: 0
  Register range end: 0
  Identifier range end: 8
  Const literal range end: 21
  Literal range end: 25

   0 : CBC_DEFINE_VARS idx:1(ident)->var_ident(set)
   2 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
   4 : CBC_POP_BLOCK
   5 : CBC_CREATE_OBJECT
   6 : CBC_POP
   7 : CBC_CREATE_OBJECT
   8 : CBC_POP
   9 : CBC_CREATE_OBJECT
  10 : CBC_BRANCH_IF_FALSE_FORWARD offset:5(->15)
  12 : CBC_POST_INCR_IDENT_PUSH_RESULT idx:3(ident)->ident(x)
  14 : CBC_POP_BLOCK
  15 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
  17 : CBC_POP_BLOCK
  18 : CBC_CREATE_OBJECT
  19 : CBC_PUSH_LITERAL idx:9(lit)->string(exp)
  21 : CBC_SET_PROPERTY idx:8(lit)->string(member)
  23 : CBC_ADD_TWO_LITERALS idx:4(ident)->ident(a) idx:5(ident)->ident(b)
  26 : CBC_SET_PROPERTY idx:10(lit)->string(set)
  28 : CBC_PUSH_IDENT idx:6(ident)->ident(c)
  30 : CBC_SET_PROPERTY idx:11(lit)->string(get)
  32 : CBC_ASSIGN_IDENT idx:0(ident)->var_ident(get)
  34 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
  36 : CBC_POP_BLOCK
  37 : CBC_CREATE_OBJECT
  38 : CBC_PUSH_IDENT idx:4(ident)->ident(a)
  40 : CBC_BRANCH_IF_FALSE_FORWARD offset:6(->46)
  42 : CBC_PUSH_IDENT idx:5(ident)->ident(b)
  44 : CBC_JUMP_FORWARD offset:4(->48)
  46 : CBC_PUSH_IDENT idx:6(ident)->ident(c)
  48 : CBC_SET_PROPERTY idx:11(lit)->string(get)
  50 : CBC_PUSH_IDENT idx:7(ident)->ident(value)
  52 : CBC_SET_PROPERTY idx:10(lit)->string(set)
  54 : CBC_ASSIGN_IDENT idx:1(ident)->var_ident(set)
  56 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
  58 : CBC_POP_BLOCK
  59 : CBC_CREATE_OBJECT
  60 : CBC_PUSH_LITERAL idx:21(lit)->function
  62 : CBC_SET_PROPERTY idx:12(lit)->string(function)
  64 : CBC_PUSH_LITERAL idx:22(lit)->function
  66 : CBC_EXT_SET_GETTER idx:11(lit)->string(get)
  69 : CBC_PUSH_LITERAL idx:23(lit)->function
  71 : CBC_EXT_SET_SETTER idx:10(lit)->string(set)
  74 : CBC_PUSH_LITERAL idx:24(lit)->function
  76 : CBC_EXT_SET_GETTER idx:11(lit)->string(get)
  79 : CBC_POP_BLOCK
  80 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
  82 : CBC_POP_BLOCK
  83 : CBC_CREATE_OBJECT
  84 : CBC_PUSH_LITERAL idx:14(lit)->number(0x09Af)
  86 : CBC_SET_PROPERTY idx:13(lit)->number(.3)
  88 : CBC_PUSH_LITERAL idx:16(lit)->number(14.)
  90 : CBC_SET_PROPERTY idx:15(lit)->number(12e-1)
  92 : CBC_CREATE_OBJECT
  93 : CBC_ADD_TWO_LITERALS idx:19(lit)->number(0.077) idx:20(lit)->number(9074)
  96 : CBC_SET_PROPERTY idx:18(lit)->number(0e+3)
  98 : CBC_SET_PROPERTY idx:17(lit)->number(0XFaBe0189)
 100 : CBC_POP_BLOCK
 101 : CBC_PUSH_IDENT idx:2(ident)->ident(next_statement)
 103 : CBC_POP_BLOCK
 104 : CBC_RETURN_WITH_UNDEFINED

Byte code size: 105 bytes

Script parsing successfully completed. Total byte code size: 109 bytes

--- Script parsing end ---

